src/force-move.ts

```diff
@@ -0,0 +1,229 @@
+// @ts-ignore
+import ForceMoveArtifact from '../build/contracts/ForceMove.json';
+import * as ethers from 'ethers';
+import {TransactionRequest} from 'ethers/providers';
+import {State, hashState, getVariablePart, getFixedPart, hashAppPart} from './state';
+import {Signature} from 'ethers/utils';
+import {hashOutcome} from './outcome';
+import {encodeChannelStorageLite} from './channel-storage';
+
+// TODO: Currently we are setting some arbitrary gas limit
+// to avoid issues with Ganache sendTransaction and parsing BN.js
+// If we don't set a gas limit some transactions will fail
+const GAS_LIMIT = 3000000;
+
+const ForceMoveContractInterface = new ethers.utils.Interface(ForceMoveArtifact.abi);
+
+export function createRespondWithAlternativeTransaction(
+  challengeState: State,
+  finalizesAt: string,
+  states: State[],
+  signatures: Signature[],
+  whoSignedWhat: number[],
```

Comment:
I'm wondering whether we can push the calculation of this into this method. As a wallet provider, I think maybe I want the interface to be something like:
```ts
function createRespondWithAlternativeTransaction(channelStorage: ChannelStorage, signedStates: SignedState[])

// with 
interface ChannelStorage {
  challengeState: State,
  finalizesAt: string,
  turnNumRecord: number,
  challenger: Address,
}
// and something like
interface SignedState {
   state: State,
   signature: Signature, // signature of mover
   additionalSignatures?: Signature[] // or maybe these should just be handled as separate SignedStates
}
```
That way I can just pull the `channelStorage` and the `signedStates` straight out of my db and not have to think about it. Maybe the method could optimize e.g. by deciding which states can be dropped if there are some states with `additionalSignatures`. It might also be that we ultimately end up with a single `createResponse`, which decides whether to respond or respond-from-alternative, depending on the states that were passed in.